### PR TITLE
Treat -1 as default value for memory swappiness

### PR DESCRIPTION
**Pull request by ktraghavendra**
As mentioned in #639 for some older kernels setting swappiness
fails. This happens even when nobody tries to configure swappiness
from docker UI because we would still get some default value from
host config.
With this we treat -1 value as default value and skip the enforcement
of swappiness.

Signed-off-by: Raghavendra K T raghavendra.kt@linux.vnet.ibm.com

### CODE REVIEW

File: cgroups/fs/memory.go [hunk 1]
```diff
@@ -75,6 +75,10 @@ func (s *MemoryGroup) Set(path string, cgroup *configs.Cgroup) error {
 		if err := writeFile(path, "memory.swappiness", strconv.FormatInt(cgroup.MemorySwappiness, 10)); err != nil {
 			return err
 		}
+	} else if cgroup.MemorySwappiness == -1 {
+		return nil
+	} else {
+		return fmt.Errorf("Invalid value:%d. Valid memory swappiness range is 0-100", cgroup.MemorySwappiness)
```
Comment (vmarmol) [inline cgroups/fs/memory.go]: nit: Go errors start with lowercase. Here and below.

File: cgroups/systemd/apply_systemd.go [hunk 2]
```diff
@@ -487,6 +487,10 @@ func setKernelMemory(c *configs.Cgroup) error {
 		if err != nil {
 			return err
 		}
+	} else if c.MemorySwappiness == -1 {
```
Comment (vmarmol) [inline cgroups/systemd/apply_systemd.go]: How is this related to the KernelMemory limit?

**Comment by LK4D4** [comment]
I think you should port this PR to opencontainers/runc/libcontainer.

File: cgroups/systemd/apply_systemd.go (hunk 2 above)
Comment (ktraghavendra) [inline cgroups/systemd/apply_systemd.go]: @vmarmol 
Thanks for the review
corrected both error format and the patching problem (KernelMemLimit) that happened while porting from the docker environment to libcontainer

**Comment by ktraghavendra** [comment]
@LK4D4 Thanks, okay will lok into opencontainers' porting
